go-app.js

```diff
@@ -0,0 +1,138 @@
+        return new FreeText(name, {
+            question: 'What would you like to put?',
+
+            next: function(content) {
+                return self
+                    .http.put('http://httpbin.org/put', {
+                        data: {message: content}
+                    })
+                    .then(function(resp) {
+                        return {
+                            name: 'states:done',
+                            creator_opts: {
+                                method: 'put',
+                                echo: resp.data.json.message
+                            }
+                        };
+                    });
+            }
+        });
+    });
+
+    self.states.add('states:post', function(name) {
+        // Similarly to the put requests above, we send it to httpbin.org, then
+        // tell the interaction machine to go to 'states:done', giving it the
+        // method that was performed and httpbin.org's echo of the content.
+        return new FreeText(name, {
+            question: 'What would you like to post?',
+
+            next: function(content) {
+                return self
+                    .http.post('http://httpbin.org/post', {
+                        data: {message: content}
+                    })
+                    .then(function(resp) {
+                        return {
+                            name: 'states:done',
+                            creator_opts: {
+                                method: 'post',
+                                echo: resp.data.json.message
+                            }
+                        };
+                    });
+            }
+        });
+    });
+
+    self.states.add('states:done', function(name, opts) {
+        // Here we use the options given in 'states:put' and 'states:post' to
+        // show the appropriate message.
+        return new EndState(name, {
+            text: [
+                "You just performed a " + opts.method + ".",
+                "It was echoed back: " + opts.echo
+            ].join(' '),
+            next: 'states:start'
+        });
+    });
+
+    self.states.add('states:error', function(name) {
+        // When we get a response with a status code that is not in the 200
+        // range, an error is thrown. We use `.catch()` (the promise method to
+        // use for error callbacks) to handle the error instead of `.then()`
+        // (the promise method to use for success callbacks).  The error object
+        // given as a parameter to `.catch()` is an HttpResponseError instance,
+        // so we can access the response off the object as `.response`. Note
+        // that if we wanted to have both success and error callbacks, we could
+        // provide both as arguments to `.then()` instead of using `.catch()`.
+        return self
+            .http.get('http://httpbin.org/status/418')
+            .catch(function(e) {
```

Comment:
I think maybe this is a bit of a strange use case. Maybe a better example would be to show using both arguments to `.then()`, since most times we want to make an http request, and do something with the data, and if the http request fails, then we want to handle the failure.

So maybe for this state, if the request passes, send them back telling that the error request passed, and if it fails, send them back that the error request failed with the following code.